specs/vm/instruction_set.md

```diff
@@ -1785,3 +1786,101 @@ Panic if:
 - `$fp->$fp == 0` (if parent context is external)
 
 Set `$rA` to `$fp->$fp` (i.e. `$rA` will point to the previous call frame's contract ID).
+
+### GTF: Get transaction fields
+
+|             |                         |
+|-------------|-------------------------|
+| Description | Get transaction fields. |
+| Operation   | Varies (see below).     |
+| Syntax      | `gtf $rA, $rB, imm`     |
+| Encoding    | `0x00 rA rB i i`        |
+| Notes       |                         |
+
+Get [fields from the transaction](../protocol/tx_format.md#transaction).
+
+| name                                      | `imm`   | set `$rA` to                                     |
+|-------------------------------------------|---------|--------------------------------------------------|
+| `GTF_TYPE`                                | `0x001` | `tx.type`                                        |
+| `GTF_SCRIPT_GAS_PRICE`                    | `0x002` | `tx.gasPrice`                                    |
+| `GTF_SCRIPT_GAS_LIMIT`                    | `0x003` | `tx.gasLimit`                                    |
+| `GTF_SCRIPT_MATURITY`                     | `0x004` | `tx.maturity`                                    |
+| `GTF_SCRIPT_SCRIPT_LENGTH`                | `0x005` | `tx.scriptLength`                                |
+| `GTF_SCRIPT_SCRIPT_DATA_LENGTH`           | `0x006` | `tx.scriptDataLength`                            |
+| `GTF_SCRIPT_INPUTS_COUNT`                 | `0x007` | `tx.inputsCount`                                 |
+| `GTF_SCRIPT_OUTPUTS_COUNT`                | `0x008` | `tx.outputsCount`                                |
+| `GTF_SCRIPT_WITNESSES_COUNT`              | `0x009` | `tx.witnessesCount`                              |
+| `GTF_SCRIPT_RECEIPTS_ROOT`                | `0x00A` | Memory address of `tx.receiptsRoot`              |
```

Comment:
This offset is constant. Is it worthy to add an operation to fetch it?

Reply:
I guess this wasn't the clearest, but the purpose of these instructions is to allow e.g. the stdlib to be able to get tx fields regardless of if additional fields are inserted and break the tx format. https://github.com/FuelLabs/fuel-specs/issues/287#issuecomment-1071890003

Without these instructions for example, we can't remove the byte price field without breaking the stdlib, which would be Very Bad for immutable smart contracts.

Reply:
I see! Maybe add a tag to the compiler a flag on whether the user wants to compile for full gas optimization (so these opcodes are translated into hardcoded constants), or for long-term compatibility.

cc @mohammadfawaz @otrho @sezna